sklearn/ensemble/_forest.py

```diff
@@ -396,7 +397,15 @@ def fit(self, X, y, sample_weight=None):
             self.estimators_.extend(trees)
 
         if self.oob_score:
-            self._set_oob_score(X, y)
+            y_type = type_of_target(y)
+            if y_type in ("multiclass-multioutput", "unknown"):
+                raise ValueError(
+                    f"The type of target cannot be used to compute OOB "
+                    f"estimates. Got {y_type} while only the following are "
+                    f"supported: continuous, continuous-multioutput, binary, "
+                    f"multiclass, multilabel-indicator."
+                )
```

Comment:
Maybe we could add a comment in the code that we could consider support the "multiclass-multioutput" case if we add a `scoring` constructor parameter to all the user to pass there own scorer for this case in particular.

Note that both multilabel classification and multioutput regression are examples of tasks where our scorers average the individual tasks scores to get a final scores. Currently our scorers refuse to perform such a naive averaging for the multiclass multioutput classification case but this rejection could be considered arbitrary since we do it for multilabel classification and multioutput regression.